test/hotspot/jtreg/compiler/c1/TestNullArrayClone.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8302850
```

Comment:
please add @summary
is the purpose of the test to check that array clone throws NPE for null input and does not throw otherwise?
Don't we want to check the contents of the copied data?
Don't we want to check different sizes and array types?
Is 1K iterations enough to compile the method?

Reply:
> please add @summary

+1

> is the purpose of the test to check that array clone throws NPE for null input and does not throw otherwise?

I added this test to verify that null values for the array where handled correctly when the array clone call had been C1 compiled. I added this test at the time because I discovered a bug in the implementation and I had not seen any existing tests fail.

> Don't we want to check the contents of the copied data?
> Don't we want to check different sizes and array types?

Verifying the contents sounds good.

Different sizes and types sounds good as well, but what sizes would you choose? For the types, I would limit it to primitive types since the intrinsic is only implemented for those.

> Is 1K iterations enough to compile the method?

That seemed to be enough in my case to trigger the issue, what number should I use instead? I found `Tier3CompileThreshold` that is 2000, so maybe increase it to that just in case? Is there a way to verify from the test that the method has been C1 compiled before going ahead and invoking the method with null?

Reply:
Added brief test summary.

Reply:
Thank you!

> what sizes would you choose? For the types, I would limit it to primitive types

Yes, checking the primitive types is fine. Let it be int[], long[], and byte[]. For size I would pick an odd.

```
static final int ITER = 2000; // ~ Tier3CompileThreshold
static final int ARRAY_SIZE = 999;
```

Reply:
If you like, you can inspect the output of the -XX:+PrintLIR option to see if C1 applies arraycopy as expected.

Reply:
Expanded the tests in this class. I verified with `PrintLIR` that the methods are being C1 compiled.